libraries/AP_NMEA_Output/AP_NMEA_Output.cpp

```diff
@@ -137,9 +145,16 @@ void AP_NMEA_Output::update()
     snprintf(gga_end, sizeof(gga_end), "*%02X\r\n", (unsigned) _nmea_checksum(gga));
 
     // get speed
-    Vector2f speed = ahrs.groundspeed_vector();
-    float speed_knots = speed.length() * M_PER_SEC_TO_KNOTS;
-    float heading = wrap_360(degrees(atan2f(speed.x, speed.y)));
+    float speed_knots = 0;
+    float heading = 0;
+    if (ahrs != nullptr) {
+        const Vector2f speed = ahrs->groundspeed_vector();
+        speed_knots = speed.length() * M_PER_SEC_TO_KNOTS;
+        heading = wrap_360(degrees(atan2f(speed.x, speed.y)));
```

Comment:
is this correct ? normally atan2f is atan2f(y, x) . I know it was like that before and I was about to fix it. Could you check that the heading report is correct ?

Reply:
ping @magicrub